### ISSUE TITLE
Test in CI that we're in sync with problem-specifications

### ISSUE THREAD

**Issue by senekor** [exercism/rust#1721]
...but we don't want our CI to break by upstream changes.

So we pin a version of problem-specifications as a submodule. The updating workflow would then be: Pull the submodule, run the test generator over all implemented exercises, fix what's broken (hopefully all caught by CI), and open a PR.

This is blocked by `configlet` being able to use a pinned version of problem-specifications, see https://github.com/exercism/configlet/issues/816.

Probably in a later step, we could add a CI cron job to check for upstream changes and open an issue (or even a PR with the update already done!)

**Comment by senekor** [comment]
Reading the linked issue in the configlet repo again, we're not really blocked. If there is no movement on the configlet side, we can do it ourselves with little effort.

The only thing that's missing now is for the repo to actually be in sync with problem-specifications. Still a bunch of exercises left to sync.

**Comment by senekor** [comment]
Maybe this should be split up into two conceptual tests:

- [ ] `configlet sync` is happy
- [ ] running the test generator over all exercises that use it is a noop

Importantly, the second point can be implemented already. This doesn't need to wait until all exercises are synced / uplifted to the test generator. Maybe it doesn't even make sense to use the test generator for some exercises that don't come from problem-specifications.
